src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,12 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period").append(dateFrom).append(" - ")
+                .append(dateTo).append("\n");
+        for (int i = 0; i < names.length; i++) {
+            stringBuilder.append(names[0]).append(" - ").append(data[0]).append("\n");
```

Comment:
The loop is intended to iterate over each name in the `names` array, but it incorrectly uses `names[0]` and `data[0]` in the `append` method. This results in the same name and data being appended repeatedly. You should use `names[i]` and the corresponding data for each name to ensure the correct information is appended for each person.